Update Fleet server config documentation for local storage (#50489)
fleetdm/fleet: docs/Configuration/fleet-server-configuration.md

## S3

> > If you're hosting Fleet on Render (or any service that doesn't offer S3-compatible storage) or building Fleet locally, use the `FLEET_SOFTWARE_INSTALLER_STORE_DIR` environment variable (no YAML or flag equivalent) to store software installers and bootstrap packages on local disk instead. If you're hosting Fleet in AWS use S3 (below) because local storage won't work for multi-container deployments.

Reviewer: ```suggestion
> If you're hosting Fleet on Render (or any service that doesn't offer S3-compatible storage) or building Fleet locally, use the `FLEET_SOFTWARE_INSTALLER_STORE_DIR` environment variable (no YAML or flag equivalent) to store software installers and bootstrap packages on local disk instead. If you're hosting Fleet in AWS, use S3 (below) because local storage won't work for multi-container deployments.
```

> > For single-container environments — like local development, or services such as Render that don't offer S3-compatible storage — Fleet can store software installers and bootstrap packages on local disk instead, via the `FLEET_SOFTWARE_INSTALLER_STORE_DIR` environment variable (no YAML or flag equivalent). This won't work correctly once you scale to more than one Fleet server container. Use S3 (below) for multi-container production deployments.

Reviewer: ```suggestion
> If you're hosting Fleet on Render (or any service that doesn't offer S3-compatible storage) or building Fleet locally, use the `FLEET_SOFTWARE_INSTALLER_STORE_DIR` environment variable (no YAML or flag equivalent) to store software installers and bootstrap packages on local disk instead. If you're hosting Fleet in AWS use S3 (below) because local storage won't work for multi-container deployments.
```

Reviewer: @marko-lisica tried to shorten and clarify the language.

What do you think?

Author: @noahtalerman Looks good to me.

## S3

> > If you're hosting Fleet on Render (or any service that doesn't offer S3-compatible storage) or building Fleet locally, use the `FLEET_SOFTWARE_INSTALLER_STORE_DIR` environment variable (no YAML or flag equivalent) to store software installers and bootstrap packages on local disk instead. If you're hosting Fleet in AWS, use S3 (below) because local storage won't work for multi-container deployments.

Reviewer: FYI @BCTBB or @rfairburn please let us know if this is wrong

Reviewer 2: The not working for multi-container deployments is 100% true, but I'd not limit it to AWS in the phrasing.   I'd make it part of the If as follows:

```suggestion
> If you're hosting Fleet on Render (or any service that doesn't offer S3-compatible storage and is not a multi-container deployment) or building Fleet locally, use the `FLEET_SOFTWARE_INSTALLER_STORE_DIR` environment variable (no YAML or flag equivalent) to store software installers and bootstrap packages on local disk instead. If you're hosting Fleet in AWS, GCP or any other large-scale deployment, use an S3 compatible object store (below) because local storage won't work for multi-container deployments.
```
